Projects/BricKuber/brickuber_lib.py

```diff
@@ -43,54 +56,73 @@ def __init__(self, robot_style, debug = False):
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
+            
+            spin_direction = 1             
+            
         elif robot_style == "EV3":
             self.TurnTablePinion = 12
             self.TurnTableGear = 36
             
             # motor position constants (these likely need to be adjusted)
-            self.MOTOR_GRAB_POSITION_HOME      = 0
-            self.MOTOR_GRAB_POSITION_REST      = -35
-            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -90
-            self.MOTOR_GRAB_POSITION_GRAB      = -130
-            self.MOTOR_GRAB_POSITION_FLIP      = -240
+            self.MOTOR_GRAB_POSITION_HOME      = -337 # -120 # 
+            self.MOTOR_GRAB_POSITION_REST      = -337 # -240 # 
+            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -280 # -280 # 
+            self.MOTOR_GRAB_POSITION_GRAB      = -240 # -306 # 
+            self.MOTOR_GRAB_POSITION_FLIP      = -120 # -393 # 
             
             self.MOTOR_GRAB_SPEED_GRAB = 400
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
         else:
             raise ValueError("Unsupported robot style")
-        
+                
         self.BP.set_motor_limits(self.MOTOR_PORTS[self.MOTOR_TURN], 0, ((500 * self.TurnTableGear) / self.TurnTablePinion))
         
         self.home_all()
     
+    def read_encoder(self):
+        encoder = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
+        debug_motor_commands("Encoder: " + str(encoder))
+    
     # find motor home positions for all motors
     def home_all(self):
         self.BP.set_motor_power(self.MOTOR_PORTS[self.MOTOR_GRAB], 15)
+        self.read_encoder()
         EncoderLast = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
         time.sleep(0.1)
         EncoderNow = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
         while EncoderNow != EncoderLast:
             EncoderLast = EncoderNow
             time.sleep(0.1)
             EncoderNow = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
-        self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow - 25))
+        if robot_style == "EV3":       # Need to revers the spin direction for the EV3 to work.    
+            self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow - 25))
+        else:
+            self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow + 25))
         self.BP.set_motor_position(self.MOTOR_PORTS[self.MOTOR_GRAB], self.MOTOR_GRAB_POSITION_REST)
         
         self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_TURN], self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_TURN]))
         self.TurnTableTarget = 0
         self.spin(0)
     
     # run a motor to the specified position, and wait for it to get there
-    def run_to_position(self, port, position, tolerance = 3):
```

Comment:
Did you change to a default of 5 degrees tolerance due to 3 degrees not working? 3 seemed to work fine for me.

Reply:
I'll change it back.  this was part of my troubleshooting; because the code had no documentation and I was on my own, I tried to change a lot of things.